resolversrv/views.py

```diff
@@ -0,0 +1,470 @@
+        return '/'.join(slashparts[:3]) + '/'
+
+
+    def __get_url_hostname(self, url):
+        """
+
+        :param url:
+        :return: the hostname only, which is used to update some of the urls from db
+        """
+        return self.__get_url_hostname_with_protocol(url).split('://', 1)[-1].rstrip('/')
+
+
+    def __get_data_source_title_url(self, link_sub_type, default_url):
+        """
+        for link type DATA we include the source url and title first and under it list the resources for requested bibcode
+        this would get the list from config and returns the info if found, otherwise builds a generic title
+
+        :param link_sub_type:
+        :param default_url:
+        :return:
+        """
+        data_resources = current_app.config['RESOLVER_DATA_SOURCES']
+        if link_sub_type in data_resources:
+            title = data_resources[link_sub_type].get('name','')
+            url = data_resources[link_sub_type].get('url','')
+        else:
+            title = 'Resource at ' + default_url
+            url = default_url
+        return [title,url]
+
+
+    def __update_data_type_hostname(self, data_source_url, link_sub_type, link_url):
+        """
+        update link_url with hostname from data source url for particular links sub types
+
+        :param data_source_url: data source url for the link sub type
+        :param link_sub_type:
+        :param link_url:
+        :return:
+        """
+        hostname = self.__get_url_hostname(data_source_url)
+        if (link_sub_type == 'NED'):
+            return link_url.replace('$NED$', hostname)
+        if (link_sub_type == 'SIMBAD'):
+            return link_url.replace('$SIMBAD$', hostname)
+        if (link_sub_type == 'Vizier') or (link_sub_type == 'CDS'):
+            return link_url.replace('$VIZIER$', hostname)
+        return link_url
+
+
+    def __return_response(self, results, status):
+        """
+        
+        :param results: results in a dict
+        :param status: status code
+        :return: 
+        """
+        response = json.dumps(results)
+
+        current_app.logger.info('sending response status={status}'.format(status=status))
+        current_app.logger.debug('sending response text={response}'.format(response=response))
+
+        r = Response(response=response, status=status)
+        r.headers['content-type'] = 'application/json'
+        return r
+
+
+    def __get_link_type_count(self, link_type):
+        """
+
+        :param link_type:
+        :return:
+        """
+        # on the fly links are only 1
+        if link_type in self.on_the_fly.keys():
+            return 1
+        # query db
+        results = get_records(bibcode=self.bibcode, link_type=link_type)
+        if len(results) == 0:
+            return 0
+        try:
+            count = 0
+            for result in results:
+                for idx in range(len(result['url'])):
```

Comment:
potential double counting, perhaps you wanted just:

```python
for u in result['url']:
    count += max(1, len(u))
```

Reply:
no actually I wanted to sum up the itemcount. thank you for pointing this out.